test/unit/specs/dashboard/cd-dashboard-children.spec.js

```diff
@@ -0,0 +1,119 @@
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+
+  describe('computed', () => {
+    describe('computed.children', () => {
+      it('should return a list containing the current user if isSingle', () => {
+        // ARRANGE
+        const vm = vueUnitHelper(DashboardChildrenComponentWithMocks);
+        vm.userChildren = [{ id: '1' }];
+
+        // ASSERT
+        expect(vm.children).to.deep.equal([{ id: '1' }]);
+      });
+      it('should return a empty list if the current user is not isSingle', () => {
+        // ARRANGE
+        const vm = vueUnitHelper(DashboardChildrenComponentWithMocks);
+        vm.userChildren = [];
+
+        // ASSERT
+        expect(vm.children).to.deep.equal([]);
+      });
+    });
+  });
+
+  describe('methods', () => {
+    describe('methods.loadCurrentUser', () => {
+      it('should load the current user', async () => {
+        // ARRANGE
+        const mockUser = {
+          id: 'foo',
+        };
+        MockUsersService.getCurrentUser.returns(Promise.resolve({ body: { user: mockUser } }));
+        const vm = vueUnitHelper(DashboardChildrenComponentWithMocks);
+
+        // ACT
+        await vm.loadCurrentUser();
+
+        // ASSERT
+        expect(vm.currentUser).to.deep.equal(mockUser);
+      });
+    });
+
+    describe('methods.loadProfile', () => {
+      it('should load the current user\'s profile', async () => {
+        // ARRANGE
+        const mockUserProfile = {
+          id: '1',
+        };
+        MockUsersService.userProfileData.returns(Promise.resolve({ body: mockUserProfile }));
+        const vm = vueUnitHelper(DashboardChildrenComponentWithMocks);
+        vm.currentUser = {
+          id: '1',
+        };
+        // ACT
+        await vm.loadProfile();
+
+        // ASSERT
+        expect(vm.userProfile).to.equal(mockUserProfile);
+      });
+    });
+
+    describe('methods.loadChildren', () => {
+      it('should load the current user\'s children', async () => {
+        // ARRANGE
+        const mockChild = {
+          userId: '1',
+          user: { id: '1' },
+        };
+        MockUsersService.userProfileData.returns(Promise.resolve({ body: mockChild }));
+        const vm = vueUnitHelper(DashboardChildrenComponentWithMocks);
+        vm.userProfile = {
+          id: '34',
+          children: ['1'],
+        };
+        // ACT
+        await vm.loadChildren();
+
+        // ASSERT
+        expect(vm.userChildren).to.deep.equal([mockChild]);
+      });
+    });
+  });
+
+  describe('created', () => {
+    it('should add a default child if the user is having no child and is not single', async () => {
+      const vm = vueUnitHelper(DashboardChildrenComponentWithMocks);
+      vm.loadCurrentUser = sinon.stub().resolves();
+      vm.loadProfile = sinon.stub().resolves();
+      vm.loadChildren = sinon.stub().resolves();
+      vm.currentUser = null;
+      vm.userProfile = {};
+      vm.userChildren = [];
+
+      await vm.$lifecycleMethods.created();
```

Comment:
Missing test? What do we assert here?

Reply:
errr is it that it actually loads the children if they have kids and doesn't if none?

Reply:
It doesn't test anything, it stubs data, but there is no "expect".
You would expect here (and the test name is wrong) that each of the component methods are being called to load the data